### ISSUE TITLE
False identification of query placeholders within strings

### ISSUE THREAD

**Issue by lanzz** [gforcada/flake8-pep3101#3]
``` python
query = """
    SELECT some_function(
        %(foo_id)s
    )
    FROM table
    """
query_params = {
    'foo_id': 123,
}
cursor.execute(query, query_params)
```

Flake8-pep3101 will complain about the percent sign in the string, but there's no remedy that:
- I can't replace the percent signs with anything else
- I cannot embed a `# noqa` within a multiline string
- I don't want to disable this check for the entire file
- I would rather not break my string just for the sake of an overeager linter

Weirdly, if I do `SELECT some_function(%(foo_id)s)` instead of having the parameter on a new line, the linter will not complain; this is not a suitable solution though, as my actual query is much more complex than that and the function call has a lot more arguments.

**Comment by The-Compiler** [comment]
I have a similar problem, with an error flagged inside a jinja template:

```
    STYLESHEET = """
        QLabel {
            font: {{ font['keyhint'] }};
            color: {{ color['keyhint.fg'] }};
            background-color: {{ color['keyhint.bg'] }};
            padding: 6px;
            {% if config.get('ui', 'status-position') == 'top' %}
                border-bottom-right-radius: 6px;
            {% else %}
                border-top-right-radius: 6px;
            {% endif %}
        }
    """
```

**Comment by gforcada** [comment]
@lanzz @The-Compiler would you mind checking this pull request: https://github.com/gforcada/flake8-pep3101/pull/10

The proper fix would be to rewrite the whole flake8 extension to use the `ast` module https://github.com/gforcada/flake8-pep3101/issues/2 but right now I don't have the time :-/ would it be something that you could work on?

flake8-builtins is already ported to use the ast module and can be helpful as a porting reference maybe: https://github.com/gforcada/flake8-builtins/blob/master/flake8_builtins.py

**Comment by The-Compiler** [comment]
That PR seems to fix the issue for me!

Unfortunately I don't have the time to contribute currently either, sorry!

**Comment by gforcada** [comment]
@The-Compiler that's good news, I will merge that PR and we can work on the other issues together with @lanzz as well.

**Comment by The-Compiler** [comment]
Thanks for merging it! Do you plan to do a new release with that in anytime soon?

**Comment by gforcada** [comment]
@The-Compiler thanks for pointing it out, the release is out: 0.6! https://pypi.org/project/flake8-pep3101/

**Comment by gforcada** [comment]
@lanzz would you mind testing the new release and commenting if this issue can  be closed or not?

**Comment by gforcada** [comment]
@lanzz @The-Compiler I just released version 1.0 with a complete rewrite using the ``ast`` module.

Please, give it a try and let me know if your problems do persist. Thanks!

**Comment by The-Compiler** [comment]
Yay, thanks! Seems to work fine for me.
